File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/player/PlayerNetworkService.java
package top.likoslupus.cellulosesz.api.player;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;

import java.net.InetAddress;
import java.util.Optional;

/**
 * Exposes only the current connection address and never performs DNS resolution.
 */
public interface PlayerNetworkService {

    Optional<InetAddress> address(CellPlayer player);

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/player/PlayerNetworkService.kt
package top.likoslupus.cellulosesz.api.player

import top.likoslupus.cellulosesz.api.platform.CellPlayer
import java.net.InetAddress

/**
 * Exposes only the current connection address and never performs DNS resolution.
 */
public interface PlayerNetworkService {

    public fun address(player: CellPlayer): InetAddress?

}
